pkg/operator/cluster/ceph/osd/pod.go

```diff
@@ -179,13 +267,21 @@ func (c *Cluster) osdContainer(devices []rookalpha.Device, selection rookalpha.S
 		devMountNeeded = true
 	}
 
+	dataVolume := v1.VolumeMount{Name: k8sutil.DataDirVolume, MountPath: k8sutil.DataDir}
+	if config.StoreConfig.StoreType == osdcfg.Filestore {
+		prop := v1.MountPropagationBidirectional
+		dataVolume.MountPropagation = &prop
+		privileged = true
+	}
 	volumeMounts := []v1.VolumeMount{
-		{Name: k8sutil.DataDirVolume, MountPath: k8sutil.DataDir},
+		dataVolume,
 		k8sutil.ConfigOverrideMount(),
 	}
 	if devMountNeeded {
 		devMount := v1.VolumeMount{Name: "devices", MountPath: "/dev"}
 		volumeMounts = append(volumeMounts, devMount)
+		udevMount := v1.VolumeMount{Name: "udev", MountPath: "/run/udev"}
```

Comment:
Can you explain `/run/udev/` is being exposed? I don't think this is a good idea. We tried already and it always ended up poorly. Also maintaining a 'link' with `udev` is not necessarily needed.

Reply:
udevadm needs to access host's /run/udev to get device info, otherwise, the udev info inside the container's /run/udev is very limited.